### PR TITLE
`openssl` exit status is "wrong" on non-macOS platforms, fix code to work anyway

### PR DESCRIPTION
**Pull request by janpio**
The old code 
```
success = system(command.join(' '))
UI.crash!("Error decrypting '#{path}'") unless success
``` 
depends on the `command` (which is `openssl ...` in this case) to correctly return an "exit status" to initate the expected crash when decrypting fails:

> system returns true if the command gives zero exit status, false for non zero exit status. Returns nil if command execution fails.
https://ruby-doc.org/core-2.4.0/Kernel.html#method-i-system

For the macOS implementation of `openssl` this seems to be true. 
Unfortunately, it is not for the Ubuntu/WSL implementation where even if decryption fails, the exit status is `true` (but an error output is generated, which `system` didn't surface.).

Using `Open3.capture3` enables us to look at or for that error output and "fix" `success` to `false` if something is returned, which indicates a failure.

(First line of PR also fixes a tiny typo)

### CODE REVIEW

**Comment by janpio** [comment]
Alternatively I could wrap this new code in a conditional `is_mac` and leave the Mac code path unchanged.

File: match/lib/match/encrypt.rb [hunk 1]
```diff
@@ -89,8 +89,16 @@ def crypt(path: nil, password: nil, encrypt: true)
       command << "-out #{tmpfile.shellescape}"
       command << "-a"
       command << "-d" unless encrypt
-      command << "&> /dev/null" unless FastlaneCore::Globals.verbose? # to show show an error message is something goes wrong
-      success = system(command.join(' '))
+      command << "&> /dev/null" unless FastlaneCore::Globals.verbose? # to show show an error message if something goes wrong
```
Comment (revolter) [inline match/lib/match/encrypt.rb]: `show` is also duplicated, if you're at it.

**Review by revolter — CHANGES_REQUESTED** [review]

**Review by revolter — APPROVED** [review]

File: match/lib/match/encrypt.rb [hunk 2]
```diff
@@ -89,8 +89,16 @@ def crypt(path: nil, password: nil, encrypt: true)
...
-      command << "&> /dev/null" unless FastlaneCore::Globals.verbose? # to show show an error message is something goes wrong
-      success = system(command.join(' '))
+      command << "&> /dev/null" unless FastlaneCore::Globals.verbose? # to show an error message if something goes wrong
+
+      _out, err, st = Open3.capture3(command.join(' '))
+      success = st.success?
+
+      # Ubuntu `openssl` does not fail on failure
+      # but at least outputs an error message
+      if err.to_s != ''
```
Comment (giginet) [inline match/lib/match/encrypt.rb]: you should use `unless err.ro_s.empty?`

**Review by giginet — APPROVED** [review]

File: match/lib/match/encrypt.rb (hunk 2 above)
Comment (janpio) [reply match/lib/match/encrypt.rb]: Thanks, I'm pretty noob in ruby. Changed it.

**Pull request merged**